WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java

```diff
@@ -10,18 +10,24 @@
 public class WPMediaUtils {
 
     // Max picture size will be 3000px wide. That's the maximum resolution you can set in the current picker.
-    private static final int OPTIMIZE_IMAGE_MAX_WIDTH = 3000;
-    private static final int OPTIMIZE_IMAGE_ENCODER_QUALITY = 85;
+    public static final int OPTIMIZE_IMAGE_MAX_WIDTH = 3000;
+    public static final int OPTIMIZE_IMAGE_ENCODER_QUALITY = 85;
 
     public static Uri getOptimizedMedia(Activity activity, SiteModel siteModel, String path, boolean isVideo) {
-        if (isVideo || !NetworkUtils.isMobileConnected(activity)) {
-            // Not on mobile data. Skip optimization.
+        if (isVideo) {
             return null;
         }
         SiteSettingsInterface siteSettings = SiteSettingsInterface.getInterface(activity, siteModel, null);
         // Site Settings are implemented on .com/Jetpack sites only
         if (siteSettings != null && siteSettings.init(false).getOptimizedImage()) {
-            String optimizedPath = ImageUtils.optimizeImage(activity, path, OPTIMIZE_IMAGE_MAX_WIDTH, OPTIMIZE_IMAGE_ENCODER_QUALITY);
+            int resizeWidth = MediaUtils.getImageWidthSettingFromString(siteSettings.getMaxImageWidth());
```

Comment:
Why isn't the resize width stored as an `int` rather than a `String`? Couldn't it simply be `-1` o `0` for original size?